KeyDoorShuffle.py

```diff
@@ -1604,7 +1615,7 @@ def validate_key_placement(key_layout, world, player):
     for code, counter in key_layout.key_counters.items():
         if len(counter.child_doors) == 0:
             continue
-        big_found = any(i.item == dungeon.big_key for i in counter.free_locations if "- Big Chest" not in i.name) or big_key_outside
+        big_found = any((i.item is not None and i.item == dungeon.big_key) or i.name == 'Hyrule Castle - Big Key Drop' for i in counter.free_locations if "- Big Chest" not in i.name) or big_key_outside
```

Comment:
pretty sure the 'Big Key Drop' isn't in the free_locations of the key counter. This line still needs to be fixed though.

Reply:
Maybe the big key drop shouldn't be in free_locations, but looking at 'create_key_counter', it looks like it will be, since it's not an important location or a dungeon event and its item does not have 'Small Key' in its name.

(Actually, now that I look at the important locations list, 'Hyrule Dungeon Cellblock' should be 'Zelda Pickup', no?)

Reply:
Stepping through the code, looks like it ends up in other_locations.  And key_layout.big_key_special can be used to detect whether the validation should check for that key drop or the free_locations.